lang/c/src/schema.c

```diff
@@ -556,6 +560,77 @@ avro_schema_record_field_append(const avro_schema_t record_schema,
 	return 0;
 }
 
+
+static int
+avro_schema_record_field_append_with_default_json(const avro_schema_t record_schema,
+                                                  const char *field_name,
+                                                  const avro_schema_t field_schema,
+                                                  const json_t *field_default)
+{
+	check_param(EINVAL, is_avro_schema(record_schema), "record schema");
+	check_param(EINVAL, is_avro_record(record_schema), "record schema");
+	check_param(EINVAL, field_name, "field name");
+	check_param(EINVAL, is_avro_schema(field_schema), "field schema");
+        check_param(EINVAL, field_default, "field_default");
```

Comment:
Instead of duplicating all of this code, you could make this function work as expected when `field_default == NULL`, and then have both of the public functions call this.  (I.e., `avro_schema_record_field_append` would become super-short, just like the new `avro_schema_record_field_append_with_default`)